codegen/src/types/mod.rs

```diff
@@ -109,6 +109,7 @@ impl<'a> TypeGenerator<'a> {
         root_mod
     }
 
+    #[allow(clippy::only_used_in_recursion)]
```

Comment:
https://rust-lang.github.io/rust-clippy/master/index.html#only_used_in_recursion

`id: u32` below is used directly in the recursive call not modified such as `id + 1` or something
I think that is fine because the recursion stuff is based on `path: Vec<String>`